pkg/addonfactory/helm_agentaddon.go

```diff
@@ -258,6 +266,17 @@ func (a *HelmAgentAddon) getDefaultValues(
 
 	if a.hostingCluster != nil {
 		defaultValues.HostingClusterCapabilities = *a.capabilities(a.hostingCluster, addon)
+	} else if a.clusterClient != nil {
+		hostingClusterName := addon.GetAnnotations()[addonapiv1alpha1.HostingClusterNameAnnotationKey]
+		if len(hostingClusterName) > 0 {
+			hostingCluster, err := a.clusterClient.ClusterV1().ManagedClusters().
+				Get(context.TODO(), hostingClusterName, metav1.GetOptions{})
+			if err != nil {
+				klog.Errorf("failed to get hostingCluster %s. err:%v", hostingClusterName, err)
```

Comment:
should we ignore err if the hostingCluster is NotFound here ?

Reply:
PTAL